Source/WebKit/Shared/API/glib/WebKitContextMenuItem.cpp

```diff
@@ -357,6 +358,31 @@ WebKitContextMenuAction webkit_context_menu_item_get_stock_action(WebKitContextM
 #endif // ENABLE(CONTEXT_MENUS)
 }
 
+/**
+ * webkit_context_menu_item_get_title:
+ * @item: a #WebKitContextMenuItem
+ *
+ * Gets the title of @item.
+ *
+ * Returns: (transfer none): the title of @item, or %NULL if @item is a separator.
+ *
+ * Since: 2.48
+ */
+const gchar* webkit_context_menu_item_get_title(WebKitContextMenuItem* item)
+{
+    g_return_val_if_fail(WEBKIT_IS_CONTEXT_MENU_ITEM(item), nullptr);
+
+#if ENABLE(CONTEXT_MENUS)
+    if (item->priv->menuItem->type() == WebCore::ContextMenuItemType::Separator)
+        return nullptr;
+    if (item->priv->titleUTF8.isNull())
+        item->priv->titleUTF8 = item->priv->menuItem->title().utf8();
+    return item->priv->titleUTF8.data();
+#else
+    g_assert_not_reached();
+#endif // ENABLE(CONTEXT_MENUS)
```

Comment:
Wouldn't it be better to assume that we always build with `ENABLE_CONTEXT_MENUS`? Only the PlayStation port disables the option.

Then we could remove the guard:


```suggestion
    if (item->priv->menuItem->type() == WebCore::ContextMenuItemType::Separator)
        return nullptr;
    if (item->priv->titleUTF8.isNull())
        item->priv->titleUTF8 = item->priv->menuItem->title().utf8();
    return item->priv->titleUTF8.data();
```

Also API functions that depend on a feature that may be disabled at build time should warn and be no-ops, instead of crashing when reached. At any rate we cannot use `g_assert_not_reached()` here.

Reply:
Yes, but this is what we currently do for all other context menu apis, so I think this patch should be consistent. Then we can decide if leave the option, but add warnings, or remove the option. I think it can be removed, I see no reason to disable it and it's a private build option in the end.

Reply:
Fair! We can make a patch later to remove `ENABLE_CONTEXT_MENUS`.